src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/AbstractCodeActionOrUserDiagnosticTest.cs

```diff
@@ -255,21 +255,31 @@ private TestParameters WithScriptOptions(TestParameters parameters)
 
         protected async Task TestMissingInRegularAndScriptAsync(
             string initialMarkup,
-            TestParameters parameters = default)
+            TestParameters parameters = default,
+            int codeActionIndex = 0)
         {
-            await TestMissingAsync(initialMarkup, WithRegularOptions(parameters));
-            await TestMissingAsync(initialMarkup, WithScriptOptions(parameters));
+            await TestMissingAsync(initialMarkup, WithRegularOptions(parameters), codeActionIndex);
+            await TestMissingAsync(initialMarkup, WithScriptOptions(parameters), codeActionIndex);
         }
 
         protected async Task TestMissingAsync(
             string initialMarkup,
-            TestParameters parameters = default)
+            TestParameters parameters = default,
+            int codeActionIndex = 0)
         {
             using (var workspace = CreateWorkspaceFromOptions(initialMarkup, parameters))
             {
                 var (actions, _) = await GetCodeActionsAsync(workspace, parameters);
                 var offeredActions = Environment.NewLine + string.Join(Environment.NewLine, actions.Select(action => action.Title));
-                Assert.True(actions.Length == 0, "An action was offered when none was expected. Offered actions:" + offeredActions);
+
+                if (codeActionIndex == 0)
+                {
+                    Assert.True(actions.Length == 0, "An action was offered when none was expected. Offered actions:" + offeredActions);
+                }
+                else
+                {
+                    Assert.True(actions.Length <= codeActionIndex, "An action was offered at the specified index when none was expected. Offered actions:" + offeredActions);
```

Comment:
The message is the wrong way around, isn't it?

```suggestion
                    Assert.True(actions.Length <= codeActionIndex, "An action was expected at the specified index when none was offered. Offered actions:" + offeredActions);
```

Reply:
Hmm, I think the existing way is correct? The method is testing to make sure that there's no code action available at the specified index.

To address your other question - this fix makes it so we don't offer the extract local function in this scenario.

Reply:
Oh whoops, you're absolutely right!